### PR TITLE
Interfaced BorderHandler and CheckBoxHandler

### PR DESCRIPTION
**Pull request by antonfirsov**
Contributes to #4378.

### CODE REVIEW

**Review by jsuarezruiz — APPROVED** [review]

**Comment by antonfirsov** [comment]
> Commenter does not have sufficient privileges for PR 4676 in repo dotnet/maui

💔 😞

File: src/Core/src/Handlers/Border/BorderHandler.Android.cs [hunk 1]
```diff
@@ -46,9 +46,9 @@ void UpdateContent()
 				NativeView.AddView(view.ToPlatform(MauiContext));
 		}
 
-		public static void MapContent(BorderHandler handler, IBorderView border)
+		public static void MapContent(IBorderHandler handler, IBorderView border)
 		{
-			handler.UpdateContent();
+			(handler as BorderHandler)?.UpdateContent();
```
Comment (PureWeen) [inline src/Core/src/Handlers/Border/BorderHandler.Android.cs:51]: For this one I'd say that `UpdateContent` doesn't really need to be a member method.  It's not using any state on the Handler so we could just make UpdateContent static here and pass in `NativeView` and `VirtualView`

`UpdateContent(handler.NativeView, handler.VirtualView)

File: src/Core/src/Handlers/Border/BorderHandler.Windows.cs [hunk 2]
```diff
@@ -46,9 +46,9 @@ protected override ContentPanel CreateNativeView()
...
         }
 
-        public static void MapContent(BorderHandler handler, IBorderView border)
-        {
-            handler.UpdateContent();
-        }
-    }
+		public static void MapContent(IBorderHandler handler, IBorderView border)
+		{
+			(handler as BorderHandler)?.UpdateContent();
```
Comment (PureWeen) [inline src/Core/src/Handlers/Border/BorderHandler.Windows.cs:51]: Same comment here

**Review by PureWeen — CHANGES_REQUESTED** [review]
Looks good! Just a couple comments on the `UpdateContent` calls and then this should be good to go

File: src/Core/src/Handlers/Border/BorderHandler.Android.cs [hunk 3]
```diff
@@ -34,21 +34,21 @@ public override void SetVirtualView(IView view)
...
 
-		void UpdateContent()
+		static void UpdateContent(IBorderHandler handler)
 		{
-			_ = NativeView ?? throw new InvalidOperationException($"{nameof(NativeView)} should have been set by base class.");
-			_ = MauiContext ?? throw new InvalidOperationException($"{nameof(MauiContext)} should have been set by base class.");
-			_ = VirtualView ?? throw new InvalidOperationException($"{nameof(VirtualView)} should have been set by base class.");
+			_ = handler.NativeView ?? throw new InvalidOperationException($"{nameof(NativeView)} should have been set by base class.");
+			_ = handler.MauiContext ?? throw new InvalidOperationException($"{nameof(MauiContext)} should have been set by base class.");
+			_ = handler.VirtualView ?? throw new InvalidOperationException($"{nameof(VirtualView)} should have been set by base class.");
```
Comment (antonfirsov) [inline src/Core/src/Handlers/Border/BorderHandler.Android.cs:41]: This validation seems to be repeated in many of the handlers. Would it make sense to dedupe it by moving it to a utility method?

Comment (PureWeen) [reply src/Core/src/Handlers/Border/BorderHandler.Android.cs:41]: sounds good 👍

Though I think some of those aren't needed anymore and are just left over from older code. 

`VirtualView` and `NativeView` will throw exceptions if they are null when you call into them

image

Comment (antonfirsov) [reply src/Core/src/Handlers/Border/BorderHandler.Android.cs:41]: Thanks! For now it was just a question. I will address this in one of the follow-up PR-s.

**Comment by rmarinho** [comment]
@antonfirsov  we merged a rename so you will need a rebase

```
  "D:\a\_work\1\s\src\Core\src\Core.csproj" (Build target) (2:56) ->
       (CoreCompile target) -> 
         D:\a\_work\1\s\src\Core\src\Handlers\Border\BorderHandler.cs(51,3): error CS0246: The type or namespace name 'NativeView' could not be found (are you missing a using directive or an assembly reference?) [D:\a\_work\1\s\src\Core\src\Core.csproj]
         D:\a\_work\1\s\src\Core\src\Handlers\Border\BorderHandler.cs(51,29): error CS0539: 'BorderHandler.NativeView' in explicit interface declaration is not found among members of the interface that can be implemented [D:\a\_work\1\s\src\Core\src\Core.csproj]
         D:\a\_work\1\s\src\Core\src\Handlers\Border\BorderHandler.cs(14,39): error CS0535: 'BorderHandler' does not implement interface member 'IBorderHandler.NativeView' [D:\a\_work\1\s\src\Core\src\Core.csproj]
         D:\a\_work\1\s\src\Core\src\Handlers\Border\IBorderHandler.cs(17,18): error CS0109: The member 'IBorderHandler.NativeView' does not hide an accessible member. The new keyword is not required. [D:\a\_work\1\s\src\Core\src\Core.csproj]
         D:\a\_work\1\s\src\Core\src\Handlers\CheckBox\ICheckboxHandler.cs(15,18): error CS0109: The member 'ICheckBoxHandler.NativeView' does not hide an accessible member. The new keyword is not required. [D:\a\_work\1\s\src\Core\src\Core.csproj]
```

**Review by PureWeen — APPROVED** [review]

**Pull request merged**